File: Arrays/Merge Sorted Array/Python/set_based_solution.py
def remove_duplicates(nums):
    seen = set()
    index = 0

    for num in nums:
        if num not in seen:
            seen.add(num)
            nums[index] = num
            index += 1

    return index


# Example usage
nums = [0,0,1,1,1,2,2,3,3,4]
k = remove_duplicates(nums)
print(k)        # Output: 5
print(nums)     # [0,1,2,3,4,2,2,3,3,4]

File: Arrays/Merge Sorted Array/Cpp/set_based_solution.cpp
#include <bits/stdc++.h>
using namespace std;

int removeDuplicates(vector<int>& nums) {
    unordered_set<int> seen;
    int index = 0;

    for (int num : nums) {
        if (seen.find(num) == seen.end()) {
            seen.insert(num);
            nums[index++] = num;
        }
    }
    return index;
}

int main() {
    vector<int> nums = {0,0,1,1,1,2,2,3,3,4};
    int k = removeDuplicates(nums);

    cout << k << endl;
    for (int i = 0; i < k; i++) cout << nums[i] << " ";
}
